File: C_Code/dcf77_loop_filter.c
/** dcf77_loop_filter.c (TR 17-June-2015)
*
* Edited by:
*
*                  Loop filter for the usage in the PLL
*                  
*
*       y           : Input signal (result of Phase Detector)
*       *z[2]       : delays states initial/final 
*       out         : Filtered output signal
*       alpha       : 
*       beta        :
*
*/
#include "dcf77_includes.h"

float dcf77_loop_filter(float y, float *z, float alpha, float beta){	//TODO: welcher Datentyp wird erwartet?

	float out = 0;
	float up;
	float down;

	float coeffs[] = {beta, 1, 1};

	up = IIR_DF1(coeffs, z, y);

	//[up, delay] = filter([beta], [1 - 1], y(k), z*);
	down = y * alpha;
	out = up + down;

	return out;
}

File: MATLAB-Code/dcf77_loop_filter.m
% dcf77_loop_filter.m (TB 27-Mai-2015)
%
% Edited by: 
%
%                   Loop filter for the usage in the PLL
%                   
%
%       inp         : Input signal (result of Phase Detector)
%       delay       : delays states initial/final
%       out         : Filtered output signal
%       alpha       : 
%       beta        :
%

function [out, delay] = dcf77_loop_filter( inp, delay, alpha, beta)

siglen = length(inp);
out=zeros(1,siglen);

for k=1:siglen    
    [up, delay] = filter([beta], [1 -1],inp(k),delay);
    down= inp(k) * alpha;
    out(k)= up + down;
end

end